File: years/2015/01/part2.ts
import input from "./input.txt";

const lines = input.split("\n");

let floor = 0;
let i = 0;
for (const c of [...lines[0]]) {
    i++;
    if (c === "(") {
        floor++;
    } else if (c === ")") {
        floor--;
    }
    if (floor === -1) {
        throw i;
    }
}
console.log(floor);

File: years/2025/01/part2.hs
import Control.Arrow

solve = words >>> map parseCommand >>> concatMap explode >>> scanl go 50 >>> filter (== 0) >>> length
  where
    explode n = replicate (abs n) (signum n)

    go a b = (a + b) `mod` 100

parseCommand :: String -> Int
parseCommand (x : xs)
    | x == 'L' = -num
    | x == 'R' = num
  where
    num = read xs

main = readFile "input.txt" >>= (print . solve)
